File: NDPServer/src/main/java/com/slender/service/listener/EmailListener.java
package com.slender.service.listener;

import com.slender.config.manager.JsonParserManager;
import com.slender.model.message.EmailMessage;
import com.slender.service.interfase.EmailService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.amqp.rabbit.annotation.RabbitListener;
import org.springframework.stereotype.Component;

@Slf4j
@Component
@RequiredArgsConstructor
public class EmailListener {
    private final EmailService emailService;
    private final JsonParserManager jsonParser;

    @RabbitListener(queues = "captcha")
    public void sendEmail(String message){
        try {
            EmailMessage emailMessage = jsonParser.parse(message, EmailMessage.class);
            emailService.sendCaptcha(emailMessage.type(), emailMessage.toEmail());
        } catch (Exception e) {
            log.error("消息处理失败: {}", message, e);
        }
    }
}

File: NDPServer/src/main/java/com/slender/service/listener/EmailListener.kt
package com.slender.service.listener

import com.slender.config.manager.JsonParserManager
import com.slender.model.message.EmailMessage
import com.slender.service.interfase.EmailService
import org.slf4j.LoggerFactory
import org.springframework.amqp.rabbit.annotation.RabbitListener
import org.springframework.stereotype.Component

@Component
class EmailListener(
    private val emailService: EmailService,
    private val jsonParser: JsonParserManager
) {
    private val log = LoggerFactory.getLogger(EmailListener::class.java)

    @RabbitListener(queues = ["captcha"])
    fun sendEmail(message: String?) {
        try {
            val emailMessage = jsonParser.parse(message, EmailMessage::class.java)
            emailService.sendCaptcha(emailMessage.type, emailMessage.toEmail)
        } catch (e: Exception) {
            log.error("消息处理失败: {}", message, e)
        }
    }
}
